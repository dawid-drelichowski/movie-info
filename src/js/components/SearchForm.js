import React, {Component} from 'react'
import PropTypes from 'prop-types'

export default class SearchForm extends Component {
  static propTypes = {
    onSubmit: PropTypes.func
  }
  static defaultProps = {
    onSubmit: () => {}
  }
  state = {query: ''}
  onSubmit = event => {
    event.preventDefault()
    this.props.onSubmit(this.state.query)
  }
  changeQuery = event => this.setState({query: event.target.value})
  render () {
    return <form onSubmit={this.onSubmit}>
      <div className="form-row">
        <div className="col-7 col-sm-8 col-md-10 col-xl-11">
          <input
            type="search"
            value={this.state.query}
            placeholder="Enter movie title"
            onChange={this.changeQuery}
            className="form-control form-control-lg"
          />
        </div>
        <div className="col-5 col-sm-4 col-md-2 col-xl-1">
          <button className="btn btn-lg btn-success">Search!</button>
        </div>
      </div>
    </form>
  }
}
