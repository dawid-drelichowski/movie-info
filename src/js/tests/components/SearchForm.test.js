import React from 'react'
import SearchForm from 'components/SearchForm'
import renderer from 'react-test-renderer'
import {mount} from 'enzyme'

describe('Search form component', () => {
  function getRenderedComponent (onSubmit = () => {}) {
    return renderer.create(<SearchForm onSubmit={onSubmit} />)
  }

  function getMountedComponent (onSubmit = () => {}) {
    return mount(<SearchForm onSubmit={onSubmit} />)
  }

  it('input should have empty value', () => {
    expect(getRenderedComponent().toJSON()).toMatchSnapshot()
  })

  it('should have value when input changed', () => {
    const value = 'movie',
      input = getMountedComponent().find('input')
    input.simulate('change', {target: {value}})
    expect(input.props().value).toEqual(value)
  })

  it('should call "onSubmit" callback when input changed and form submitted', () => {
    const value = 'movie',
      onSubmit = jest.fn(),
      form = getMountedComponent(onSubmit),
      input = form.find('input')
    input.simulate('change', {target: {value}})
    form.simulate('submit')
    expect(input.props().value).toEqual(value)
    expect(onSubmit).toHaveBeenCalledTimes(1)
    expect(onSubmit).toBeCalledWith(value)
  })

  it('should not call "onSubmit" callback when no input changed and form submitted', () => {
    const onSubmit = jest.fn(),
      form = getMountedComponent(onSubmit)
    form.simulate('submit')
    expect(onSubmit).toHaveBeenCalledTimes(0)
  })

  it('should not call "onSubmit" callback twice when input changed once and form submitted twice', () => {
    const value = 'movie',
      onSubmit = jest.fn(),
      form = getMountedComponent(onSubmit),
      input = form.find('input')
    input.simulate('change', {target: {value}})
    form.simulate('submit')
    form.simulate('submit')
    expect(input.props().value).toEqual(value)
    expect(onSubmit).toHaveBeenCalledTimes(1)
    expect(onSubmit).toBeCalledWith(value)
  })
})
