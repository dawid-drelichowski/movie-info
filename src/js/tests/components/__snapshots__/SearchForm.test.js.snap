// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search form component input should have empty value 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-row"
  >
    <div
      className="col-7 col-sm-8 col-md-10 col-xl-11"
    >
      <input
        className="form-control form-control-lg"
        onChange={[Function]}
        placeholder="Enter movie title"
        type="search"
        value=""
      />
    </div>
    <div
      className="col-5 col-sm-4 col-md-2 col-xl-1"
    >
      <button
        className="btn btn-lg btn-success"
      >
        Search!
      </button>
    </div>
  </div>
</form>
`;
